x-pack/filebeat/input/httpjson/config_auth.go

```diff
@@ -167,6 +164,14 @@ func (o *oAuth2Config) client(ctx context.Context, client *http.Client) (*http.C
 	}
 }
 
+// maybeString returns the string pointed to by p or "" if p in nil.
+func maybeString(p *string) string {
```

Comment:
the name maybeString suggests that the type maybe a string or something else entirely suggesting we are doing a type check here, but in reality we are checking if the string is nil or not and returning a value accordingly. Maybe something like checkIfEmpty or something similar makes more sense ?

Reply:
It's not a check it's a weak maybe in the vein of a Rust maybe. It either is a string or it is not valid. In the case that it's not valid, an empty string suffices.